src/main/java/Duke.java

```diff
@@ -1,10 +1,278 @@
+                    "'event task_description /at timeframe'");
+        }
+
+        String description = request.substring(0, indexOfAt).strip();
+        String at = request.substring(indexOfAt + 4).strip();
+        storeTask(new Event(description, at));
+    }
+
+    public static void storeTask(Task taskToStore) {
+        STORED_TASKS.add(taskToStore);
+        saveDataToFile();
+        System.out.println(DIVIDER);
+        System.out.println(" Got it. I've added this task:");
+        System.out.println("   " + taskToStore);
+        System.out.println(" Now you have " + STORED_TASKS.size() + " tasks in the list.");
+        System.out.println(DIVIDER);
+    }
+
+    public static void displayStoredTasks() {
+        if (STORED_TASKS.size() == 0) {
+            System.out.println(DIVIDER);
+            System.out.println(" You have no tasks in your list! :)");
+            System.out.println(DIVIDER);
+            return;
+        }
+
+        System.out.println(DIVIDER);
+        System.out.println(" Here are the tasks in your list:");
+        for (int i = 0; i < STORED_TASKS.size(); i++) {
+            Task currentTask = STORED_TASKS.get(i);
+            System.out.printf(" %d.%s\n", (i + 1), currentTask);
+        }
+        System.out.println(DIVIDER);
+    }
+
+    public static void markTaskAsDone(String command) throws DukeException {
+        if (command.equals("done")) {
+            throw new DukeException("☹ OOPS!!! The task number of a done command cannot be empty.");
+        }
+
+        if (!command.startsWith("done ")) {
+            throw new DukeException("☹ OOPS!!! The correct syntax for done command is: 'done task_number'");
+        }
+
+        String taskNumber = command.substring(5).strip();
+        int indexOfTaskToMark;
+        try {
+            indexOfTaskToMark = Integer.parseInt(taskNumber) - 1;
+        } catch (NumberFormatException e) {
+            throw new DukeException("☹ OOPS!!! The correct syntax for done command is: 'done task_number'");
+        }
+
+        if (indexOfTaskToMark >= STORED_TASKS.size() || indexOfTaskToMark < 0) {
+            throw new DukeException("☹ OOPS!!! You entered an invalid task number!");
+        }
+
+        Task taskToMark = STORED_TASKS.get(indexOfTaskToMark);
+        taskToMark.markAsDone();
+        saveDataToFile();
+        System.out.println(DIVIDER);
+        System.out.println(" Nice! I've marked this task as done:");
+        System.out.println("   " + taskToMark);
+        System.out.println(DIVIDER);
+    }
+
+    public static void deleteTask(String command) throws DukeException {
+        if (command.equals("delete")) {
+            throw new DukeException("☹ OOPS!!! The task number of a delete command cannot be empty.");
+        }
+
+        if (!command.startsWith("delete ")) {
+            throw new DukeException("☹ OOPS!!! The correct syntax for delete command is: 'delete task_number'");
+        }
```

Comment:
Can these checks be done elsewhere?
*you may want to create more classes/methods to enable checking elsewhere.